main/src/hadd.cxx

```diff
@@ -421,7 +449,7 @@ int main( int argc, char **argv )
       exit(1);
    }
 
-   auto filesToProcess = argc - ffirst;
+   auto filesToProcess = allSubfiles.size();
```

Comment:
```suggestion
   auto nFilesToProcess = allSubfiles.size();
```
While scanning the code, I kept read `fileToProcess` has refering to the collection rather than the number of element in the collection.  To some extent it might even be better to remove this variable and sue `allSubfiles.size()` explicitly.   This allows, when reading the code, to keep in mind the correlation between the number and the corresponding collection. (and make the code more consistent as several other placed already use `allSubfiles.size()`